polipy/extractors.py

```diff
@@ -11,8 +12,10 @@ def extract(extractor, **kwargs):
         content = extract_text(**kwargs)
     return content
 
-def extract_text(url_type, url=None, dynamic_source=None, static_source=None, **kwargs):
-    if url_type is None or url_type in ['html', 'other']:
+def extract_text(url_type, url=None, dynamic_source=None, static_source=None, html_file=None, **kwargs):
```

Comment:
Looks good! I would do it this way so you can save the extracted keywords separately, as a separate extractor:

```python3
def extract(extractor, **kwargs):
    content = None
    if extractor == 'text':
        content = extract_text(**kwargs)
    elif extractor == 'keywords' and 'html_file' in kwargs and kwargs['html_file'] is not None:
        content = extract_ccpa_info(kwargs['html_file'])
    return content

def extract_text(url_type, url=None, dynamic_source=None, static_source=None, **kwargs):
    if url_type is None or url_type in ['html', 'other']:
        content = extract_html(dynamic_source, url)
    elif url_type == 'pdf':
        content = extract_pdf(static_source)
    elif url_type == 'plain':
        content = dynamic_source
    else:
        content = dynamic_source
    return content
```

Maybe take a look and see if that achieves the same functionality? (apart from not saving it under the "text" extractor).